status emoji: changed the poition of the emoji to be left of status t…

y-emam opened the pull request:
…ext or right of username

<!-- Describe your pull request here.-->
I have edited the template file that is responsible of how the items should be displayed, so it shows the status emoji to the left of the status text, and if there is no status text, it is placed to the right of the username.
It also works with and without avatars.

Fixes: <!-- Issue link, or clear description.-->
#32710 

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**
![395706785-421a3ec2-417b-4479-a23a-f153cf8b4711](https://github.com/user-attachments/assets/f63dfb66-b652-415d-ab69-453f4189e306)
![395706801-d9179a4a-fa85-4d52-9b20-c2f0757b1002](https://github.com/user-attachments/assets/5c5837af-3559-4393-92cf-645bcd87eb23)

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

whilstsomebody commented:
Hi @y-emam , as @alya mentioned in the description of the issue [#32710 ], [#32699](https://github.com/zulip/zulip/pull/32699) is already a working PR for the issue. So, if there is something in this PR that is not in the mentioned one, please write that in your PR description. 

Also, the emoji is needed to be on the left side of the status text and not of the user name.
Thanks!

y-emam commented:
@whilstsomebody thanks a lot for your feedback

y-emam commented:
Hi @whilstsomebody, 

- The status emoji's position was aligned as described in the issue; the screenshots were just outdated. I've updated the first message in the PR to reflect the correct alignment.
- I've also updated the commit message based on the contributor guide.

Let me know if you have any other feedback—I'm open to any additional suggestions!

alya commented:
Can you guys agree on which of the two PRs should be reviewed and close the other one? We can't review two PRs for the same issue.

alya commented:
You can use https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html#mentioning-other-contributors if any of the commits are collaborative.

y-emam commented:
@whilstsomebody what do you think about this?

We can keep one of the 2 PRs and we credit the other contributer in the PR.

whilstsomebody commented:
@y-emam , I have posted a comment [here](https://github.com/zulip/zulip/issues/32710#issuecomment-2546827672).

whilstsomebody commented:
Hi @y-emam , please go forward and complete the work on this PR as I want you to submit it. I will close mine.
Just few points:

- Don't forget to co-author me, I will send you my email on dm.
- Improve your commit and PR description. Take suggestions from my PR.
- Improve you code especially css.

All the best.

whilstsomebody commented:
@alya, tagging you for a maintainer review.

alya commented:
Thanks! @evykassirer could you please take a look at this one? I haven't tested.

Note from the issue:

> This is worth working on only if you can produce a really clean, easy to review PR, as we are not confident we need to make this change.

evykassirer commented:
Sorry for the delay! I'm going to come back to this after the 10.0 release work is done.

evykassirer commented on web/templates/presence_row.hbs:
```diff
@@ -4,11 +4,17 @@
+                    <div class="user-name-and-status-emoji">
+                        {{> user_full_name .}}
+                    </div>
+                    <span class="status-text"><span class="status-emoji">{{> status_emoji status_emoji_info}}</span> {{status_text}}</span>
```
This would be easier to read like this:

```
<span class="status-text">
    <span class="status-emoji">{{> status_emoji status_emoji_info}}</span>
    {{status_text}}
</span>
```

evykassirer commented on web/templates/presence_row.hbs:
```diff
@@ -4,11 +4,17 @@
             <a class="user-presence-link" href="{{href}}">
                 <div class="user-name-and-status-wrapper">
+                    {{#if status_text}}
+                    <div class="user-name-and-status-emoji">
```
`user-name-and-status-emoji` is a misleading classname now, maybe rename the classname globally to just `user-name`? Or maybe just change the classname here and have it share styles with `user-name-and-status-emoji`?

evykassirer commented on web/styles/right_sidebar.css:
```diff
@@ -416,6 +416,12 @@ $user_status_emoji_width: 24px;
         text-overflow: ellipsis;
         opacity: 0.75;
         font-size: 90%;
+        transform: translateX(-4px);
```
Do `margin-left: 0` on `.status-emoji` instead.

evykassirer commented on web/styles/right_sidebar.css:
```diff
@@ -416,6 +416,12 @@ $user_status_emoji_width: 24px;
+
+        .status-emoji {
+            position: relative;
+            top: -1px;
```
I don't think you need this position relative or position change.

evykassirer reviewed (commented):
Sorry for the very long delay! It overall looks great but left a couple comments.

timabbott commented:
Closing, since I think we've decided this change would not be an improvement to the UI. Thanks for the effort @y-emam!
